File: tests/fixtures/cpp/dirt.cpp
// ventouse C++ fixtures — Dirt (P1 effects, P2 contagion). dirty = 10 + 10*owned_lines.
#include <vector>
#include <iostream>

// CPP-A — pure arithmetic -> clean 0
int add(int a, int b) { return a + b; }

// CPP-B — IO (std::cout) -> dirty 20 (owned 1)
void log_it(int x) { std::cout << x; }

// CPP-C — input mutation through a non-const ref -> dirty 20 (owned 1)
void push_it(std::vector<int>& v, int x) { v.push_back(x); }

// CPP-C-CLEAN — const ref + const method -> cannot mutate -> clean 0
int size_of(const std::vector<int>& v) { return v.size(); }

// CPP-GLOBAL — non-const global write -> dirty ; const read -> clean.
const int MAX = 10;
int g_count = 0;
int read_max() { return MAX; } // const read -> clean 0
void bump() { g_count += 1; }  // non-const global write -> dirty 20 (owned 1)

// CPP-D — contagion (P2): wlog dirty 20, run infected (owned 2) -> 30. total 50.
void wlog(int m) { std::cout << m; }
void run(int d) {
    int r = d * 2;
    wlog(r);
}

File: tests/fixtures/rust/dirt.rs
// ventouse Rust fixtures — Dirt (P1 effects, P2 contagion). dirty = 10 + 10*owned_lines.

// RS-A — pure arithmetic -> clean 0
fn add(a: i32, b: i32) -> i32 { a + b }

// RS-B — IO (println!) -> dirty 20 (owned 1)
fn log_it(x: i32) { println!("{x}"); }

// RS-C — input mutation through a &mut param -> dirty 20 (owned 1)
fn push_it(v: &mut Vec<i32>, x: i32) { v.push(x); }

// RS-C-CLEAN — &T cannot mutate the caller (signature-precise) -> clean 0
fn sum_it(v: &Vec<i32>) -> i32 { v.iter().sum() }

// RS-GLOBAL — `static mut` write -> dirty ; `const` read -> clean.
const MAX: i32 = 10;
static mut COUNTER: i32 = 0;
fn read_max() -> i32 { MAX }           // const read -> clean 0
fn bump() { unsafe { COUNTER += 1; } } // static mut write -> dirty 20 (owned 1)

// RS-D — contagion (P2): wlog dirty 20, run infected (owned 2) -> 30. total 50.
fn wlog(m: i32) { println!("{m}"); }
fn run(d: i32) {
    let r = d * 2;
    wlog(r);
}
